lib/src/container/mod.rs

```diff
@@ -324,6 +324,23 @@ impl<'a> ManifestDiff<'a> {
         println!("Removed layers:   {n_removed:<4}  Size: {removed_size_str}");
         println!("Added layers:     {n_added:<4}  Size: {added_size_str}");
     }
+
+    /// Exports the total, removed and added content between two OCI images as a string
+    pub fn export_as_string(&self) -> String {
```

Comment:
But are we going to be *parsing* this string elsewhere?  If so, I think I'd say instead we should be returning a proper `struct`.

Reply:
I parse the string using in the following changes to rpm-ostree: https://github.com/coreos/rpm-ostree/blob/7dc4ba82849bccdeef039c6e1a826e6c03bad8ad/src/lib/rpmostree-db.c#L159-L216 (which is from: https://github.com/coreos/rpm-ostree/pull/4486).

Changing this code to return a struct would just require a few changes to the rpm-ostree PR and would be a lot cleaner compare to how I'm parsing the string. However, is there a way to convert a Rust struct to a C struct using CXX bridge?

Reply:
Yeah, so what you're hitting is generic C++/Rust integration being painful.  Really, *all* cross-language FFI is painful.

On this specific topic, there is https://cxx.rs/shared.html - however, I don't think it can be used for "foreign" types (i.e. structs in this crate), so we'd need to create another copy of the struct in rpm-ostree, and then convert it on the boundary.

I can help you with this in a realtime call if you want.

BTW what may help us here is to change the implementation of `print` in this crate to use the structs - this would avoid duplicating logic more.

Reply:
Take a look at e.g. https://github.com/coreos/rpm-ostree/blob/d498bac1dd00344b23ccb2aa8d590c34acc691e2/rust/src/lib.rs#L445 as an example and how that's bridged between Rust/C++.